### PR TITLE
Remove multi target build

### PR DESCRIPTION
**Pull request by SimonBaeumer**
## Description

Clean up multi-target builds.
I've never used the multi-target debug builds, proposing here to remove them to reduce Makefile and Dockerfile complexities.

This PR:

 - Remove `multi-target` capabilities
 - Rename `Dockerfile.hybrid` to `Dockerfile.local`
 - Remove `debug` and `build-debug` targets from Probe and Fleet-Manager Dockerfiles
 - Rename `build-standard` and to `build`
 - Rename Makefile targets
    - `image/build/multi-target/probe` to `image/build/probe`
    - `image/build/multi-target/fleet-manage` to `image/build/fleet-manager`
 - Remove `DEBUG_PODS` env var from dev scripts

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [ ] Added test description under `Test manual`
- [ ] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)
- [ ] CI and all relevant tests are passing
- [ ] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.
- [ ] Add secret to app-interface Vault or Secrets Manager if necessary
- [ ] RDS changes were e2e tested [manually](../docs/development/howto-e2e-test-rds.md)
- [ ] Check AWS limits are reasonable for changes provisioning new resources

## Test manual

**TODO:** Add manual testing efforts

```
# To run tests locally run:
make db/teardown db/setup db/migrate
make ocm/setup OCM_OFFLINE_TOKEN=<ocm-offline-token> OCM_ENV=development
make verify lint binary test test/integration
```

### CODE REVIEW

**Review by kovayur — APPROVED** [review]

**Review by vladbologa — CHANGES_REQUESTED** [review]
This PR seems to break `make deploy/dev` builds on my arm64 mac, please don't merge until I figure out what the problem is.

**Review by vladbologa — CHANGES_REQUESTED** [review]
There is a small error in `docker.sh` and the regular build is made even if `ARCH==arm64.

If I fix that, I still run into this problem that I haven't yet fixed:
```
2023-09-18T18:53:11,N:z Applying resource from '/dev/env/manifests/fleet-manager/01-fleet-manager-secrets.yaml'
secret/fleet-manager created
2023-09-18T18:53:11,N:z Applying resource from '/dev/env/manifests/fleet-manager/02-fleet-manager-deployment.yaml'
deployment.apps/fleet-manager created
2023-09-18T18:53:12,N:z Applying resource from '/dev/env/manifests/fleet-manager/03-fleet-manager-service.yaml'
service/fleet-manager created
deployment.apps/fleet-manager env updated
deployment.apps/fleet-manager env updated
deployment.apps/fleet-manager env updated
deployment.apps/fleet-manager env updated
deployment.apps/fleet-manager env updated
2023-09-18T18:53:12,N:z Waiting for container fleet-manager within pod application=fleet-manager in namespace acsms to appear...
2023-09-18T18:55:22,N:z Timed out waiting for container fleet-manager to appear for pod application=fleet-manager in namespace acsms
make: *** [deploy/dev] Error 1
```

If I switch to the main branch, the deployment works fine.

File: Makefile [hunk 1]
```diff
@@ -901,10 +863,12 @@ deploy/bootstrap:
 # Deploy local images fast for development
 deploy/dev-fast: image/build deploy/dev-fast/fleet-manager deploy/dev-fast/fleetshard-sync
```
Comment (vladbologa) [inline Makefile:864]: `make deploy/dev` now uses `image/build/local`
`make deploy/dev-fast` uses `image/build`

Is that intentional?

File: dev/env/scripts/docker.sh [hunk 2]
```diff
@@ -56,21 +56,8 @@ ensure_fleet_manager_image_exists() {
...
-                        make -C "${GITROOT}" image/build/multi-target
-                    else
-                        # We *could* also use image/build/multi-target, because that
-                        # target also supports building of standard (i.e. non-debug) images.
-                        # But until there is a reliable and portable caching mechanism for dockerized
-                        # Go projects, this would be regression in terms of build performance.
-                        # Hence we don't use the image/build/multi-target target here, but the
-                        # older `image/build/local` target, which uses a hybrid building
-                        # approach and is much faster.
                         log "Building standard image..."
```
Comment (vladbologa) [inline dev/env/scripts/docker.sh:59]:
```suggestion
                        log "Building local image..."
```

**Review by vladbologa — CHANGES_REQUESTED** [review]
I'm still having problems running the scripts. Can you please check that `make deploy/dev` works for you?

For me it times out waiting for fleet-manager to start:
```
2023-09-19T11:20:58,N:z Waiting for container fleet-manager within pod application=fleet-manager in namespace acsms to appear...
2023-09-19T11:23:08,N:z Timed out waiting for container fleet-manager to appear for pod application=fleet-manager in namespace acsms
make: *** [deploy/dev] Error 1
```

**Review by vladbologa — CHANGES_REQUESTED** [review]

File: dev/env/scripts/docker.sh [hunk 3]
```diff
@@ -57,19 +57,11 @@ ensure_fleet_manager_image_exists() {
                 # Attempt to build this image.
                 if [[ "$FLEET_MANAGER_IMAGE" == "$(make -s -C "${GITROOT}" full-image-tag)" ]]; then
                     # Looks like we can build this tag from the current state of the repository.
-                    if [[ "$DEBUG_PODS" == "true" ]]; then
-                        log "Building image with debugging support..."
-                        make -C "${GITROOT}" image/build/multi-target
+                    log "Building image..."
+                    if [[ $"ARCH" == "arm64" ]]; then
```
[the remark this answers is not in the document]
Comment (SimonBaeumer) [reply dev/env/scripts/docker.sh]: done

File: dev/env/scripts/docker.sh (hunk 2 above)
Comment (SimonBaeumer) [reply dev/env/scripts/docker.sh]: done

File: Makefile [hunk 4]
```diff
@@ -511,37 +505,29 @@ docker/login/internal:
...
 	@echo "export FLEET_MANAGER_IMAGE=$(SHORT_IMAGE_REF)"
-.PHONY: image/build/multi-target/fleet-manager
+.PHONY: image/build
 
-image/build/multi-target/probe: GOOS=linux
-image/build/multi-target/probe: IMAGE_REF="$(external_image_registry)/$(probe_image_repository):$(image_tag)"
-image/build/multi-target/probe:
-	DOCKER_CONFIG=${DOCKER_CONFIG} $(DOCKER) build --target $(IMAGE_TARGET) -t $(IMAGE_REF) -f probe/Dockerfile .
+# Builds the binaries locally and copies them into the image. This build is compatible with arm64.
+image/build/local: DOCKERFILE="Dockerfile.local"
```
[the remark this answers is not in the document]
Comment (SimonBaeumer) [reply Makefile:521]: done

File: Makefile (hunk 1 above)
Comment (vladbologa) [reply Makefile:865]:
```suggestion
deploy/dev-fast: deploy/dev-fast/fleet-manager deploy/dev-fast/fleetshard-sync
```

There was still one instance left. 😅

**Review by vladbologa — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (SimonBaeumer) [reply Makefile]: :tada: fixed

**Review by vladbologa — APPROVED** [review]

**Pull request merged**